sdk/eventhub/eventhubs-checkpointstore-table/test/tables-checkpointstore.spec.ts

```diff
@@ -0,0 +1,556 @@
+dotenv.config();
+
+/* test to show that test framework is set up well */
+describe("TableCheckpointStore", () => {
+  it("is exported from the package", () => {
+    should.exist(TableCheckpointStore, "Expected TableCheckpointStore to be exported.");
+  });
+});
+
+const service = {
+  storageConnectionString: env[EnvVarKeys.STORAGE_CONNECTION_STRING],
+  storageAccountName: env[EnvVarKeys.STORAGE_ACCOUNT_NAME],
+  storageAccountKey: env[EnvVarKeys.STORAGE_ACCOUNT_KEY],
+  tableName: env[EnvVarKeys.TABLENAME]
+};
+
+const credential = new AzureNamedKeyCredential(
+  service.storageAccountName,
+  service.storageAccountKey
+);
+const serviceClient = new TableServiceClient(
+  `https://${service.storageAccountName}.table.core.windows.net`,
+  credential
+);
+
+describe("TableCheckpointStore", function(): void {
+  let client: TableClient;
+  let table_name: string;
+
+  describe("Runs tests on table with no entities", function() {
+    table_name = `table${new Date().getTime()}`;
+    beforeEach("creating table", async () => {
+      await serviceClient.createTable(table_name);
+    });
+
+    client = new TableClient(
+      `https://${service.storageAccountName}.table.core.windows.net`,
+      table_name,
+      credential
+    );
+
+    afterEach(async () => {
+      await serviceClient.deleteTable(table_name);
+    });
+
+    describe("listOwnership", function() {
+      it("listOwnership should return an empty array", async function(): Promise<void> {
+        const checkpointStore = new TableCheckpointStore(client);
+        const listOwnership = await checkpointStore.listOwnership(
+          "test.servicebus.windows.net",
+          "testHub",
+          "testConsumerGroup"
+        );
+        should.equal(listOwnership.length, 0);
+      });
+    });
+
+    describe("updateOwnership", function() {
+      it("forwards errors", async () => {
+        const checkpointStore = new TableCheckpointStore(client);
+        const eventHubProperties = {
+          fullyQualifiedNamespace: "brown.servicebus.windows.net",
+          eventHubName: "testEventHub",
+          consumerGroup: "testConsumerGroup"
+        };
+        // now let's induce a bad failure (removing the table)
+        await serviceClient.deleteTable(table_name);
+
+        // Create the checkpoint to add.
+        const checkpoint: Checkpoint = {
+          consumerGroup: eventHubProperties.consumerGroup,
+          eventHubName: eventHubProperties.eventHubName,
+          fullyQualifiedNamespace: eventHubProperties.fullyQualifiedNamespace,
+          offset: 0,
+          partitionId: "0",
+          sequenceNumber: 1
+        };
+        try {
+          await checkpointStore.updateCheckpoint(checkpoint);
+          throw new Error("Failed");
+        } catch (err) {
+          err.message.should.not.equal("Failed");
+        }
+      });
+    });
+
+    describe("Runs tests on a populated table", function() {
+      beforeEach("creating table", async () => {
```

Comment:
Nit: "creating and populating table"

So this `beforeEach` is meant to create a table and populate it with ownership and checkpoint entities. Your tests then make assertions based on the entities that exist in the table.

I think it would help maintainability a lot if we refactored this a bit so that your tests can reference some information about the entities.

For starters, why don't we start with just a single namespace, event hub name, and consumer group? You can store these outside of the `beforeEach`, and then in each of your tests, reference these variables directly.

Right away that change gives you the benefit of not having to copy and paste the actual values in each test; you just copy and paste the variable (and the compiler will let you know if you typo that!)

Another useful thing to store would be the partition count. Right now you're creating 8 entities: 4 checkpoint and 4 ownership. There's no way to access that from your tests currently, so if you wanted to assert that the `list` operations returned the right number of results, you'd need to hard-code that. Instead, you can store the desired `partitionCount` outside of the `beforeEach`, and reference that both in your tests, and in your population logic.